File: AoC2017Tests/Day02Test.fs
module Day02Test

open AoC2017
open NUnit.Framework
open Common.Extensions

[<TestCase("day02test1", "18")>]
[<TestCase("day02", "54426")>]
let Part1 (file: string, expected : string) =
    let day = new Day02(file)
    let actual = day.SolvePart1()
    Assert.AreEqual(expected, actual)



[<TestCase("day02test2", "9")>]
let Part2 (file: string, expected : string) =
    let day = new Day02(file)
    let actual = day.SolvePart2()
    Assert.AreEqual(expected, actual)

File: AoC2017Tests/Day02Test.cs
namespace AoC2017Tests;
            
public class Day02Test
{
	[Test]
    public async Task Part1( )
    {
	    var day02 = new Day02("day02test1");
		var actual = await day02.SolvePart1( );
        Assert.That(actual, Is.EqualTo("18"));
    }
            
    [Test]
    public async Task Part2( )
    {
		var day02 = new Day02("day02test2");
		var actual = await day02.SolvePart2( );
        Assert.That(actual, Is.EqualTo("9"));
    }
}
